Add cocotb.simulator to Python library reference (#1825)
cocotb/cocotb: documentation/source/library_reference.rst

    :member-order: bysource

Simulator
---------

> 

Reviewer: Perhaps worth a note like I had in https://github.com/cocotb/cocotb/pull/1568/files#diff-7d491a84366bbf6a395ee614a5de3f21R365?

Author: Done.

Reviewer: You don't appear to have pushed

Author: I was waiting to see if I could add more, but will push now.

    :member-order: bysource

The ``cocotb.simulator`` module
-------------------------------

> This module is the interface to the GPI interface.

Reviewer: Does the `i` in gpi mean interface? Perhaps my original wording was bad, and this should be:
```suggestion
This module is a python wrapper to libgpi.
```

Reviewer: @ktbarrett, any opinions?

Reviewer 2: Looks like a bit of RAS syndrome.

Author: Done.

It should not be considered public API, but is documented here for developers
of cocotb.

.. automodule:: cocotb.simulator
    :members:
    :undoc-members:

>     :member-order: bysource

Reviewer: Note that this does not actually work, but it might do after https://github.com/sphinx-doc/sphinx/pull/7676 is merged
